kurtosis-module/impl/grafana/grafana_launcher.go

```diff
@@ -79,111 +69,58 @@ func LaunchGrafana(
 
-	datasourceTemplateData := datasourceConfigTemplateData{
+	datasourceData := datasourceConfigTemplateData{
 		PrometheusURL: prometheusPrivateUrl,
 	}
-	if err := service_launch_utils.FillTemplateToPath(datasourceConfigTemplate, datasourceTemplateData, datasourceConfigFilepath); err != nil {
-		return "", stacktrace.Propagate(err, "An error occurred filling the datasource config template")
-	}
+	datasourceTemplateAndData := enclaves.NewTemplateAndData(datasourceConfigTemplate, datasourceData)
 
-	dashboardProvidersTemplateData := dashboardProvidersConfigTemplateData{
+	dashboardProvidersData := dashboardProvidersConfigTemplateData{
 		// Grafana needs to know where the dashboards config file will be on disk, which means we need to feed
 		//  it the *mounted* location on disk (on the Grafana container) when we generate this on the module container
-		DashboardsDirpath: dashboardConfigFilepathOnGrafanaContainer,
-	}
-	if err := service_launch_utils.FillTemplateToPath(dashboardProvidersConfigTemplate, dashboardProvidersTemplateData, dashboardProvidersConfigFilepath); err != nil {
-		return "", stacktrace.Propagate(err, "An error occurred filling the dashboard config providers template")
+		DashboardsDirpath: grafanaDashboardsFilepathOnService,
 	}
+	dashboardProvidersTemplateAndData := enclaves.NewTemplateAndData(dashboardProvidersConfigTemplate, dashboardProvidersData)
 
-	if err := addGrafanaDashboardConfigToConfigDir(
-		static_files.GrafanaDashboardConfigFilepath,
-		dashboardConfigFilepath,
-	); err != nil {
-		return "", stacktrace.Propagate(
-			err,
-			"An error occurred copying Grafana dashboard config file '%v' to the Grafana config directory at '%v'",
-			static_files.GrafanaDashboardConfigFilepath,
-			dashboardConfigFilepath,
-		)
+	templateAndDataByDestRelFilepath := make(map[string]*enclaves.TemplateAndData)
+	templateAndDataByDestRelFilepath[datasourceConfigRelFilepath] = datasourceTemplateAndData
+	templateAndDataByDestRelFilepath[dashboardProvidersConfigRelFilepath] = dashboardProvidersTemplateAndData
+
+	renderedTemplateArtifactUuid, err := enclaveCtx.RenderTemplates(templateAndDataByDestRelFilepath)
+	if err != nil {
+		return "", "", stacktrace.Propagate(err, "An error occurred rendering Grafana templates")
 	}
 
-	artifactUuid, err := enclaveCtx.UploadFiles(grafanaConfigDirpathOnModule)
+	uploadArtifactUuid, err := enclaveCtx.UploadFiles(static_files.GrafanaDashboardsConfigDirpath)
 	if err != nil {
-		return "", stacktrace.Propagate(err, "An error occurred uploading Grafana config dir at '%v'", grafanaConfigDirpathOnModule)
+		return "", "", stacktrace.Propagate(err, "An error occurred uploading Grafana dashboard.json")
 	}
 
-	return artifactUuid, nil
+	return renderedTemplateArtifactUuid, uploadArtifactUuid, nil
 }
 
 func getContainerConfigSupplier(
-	configDirArtifactUuid services.FilesArtifactUUID,
+	renderTemplateArtifactUuid services.FilesArtifactUUID,
```

Comment:
Same comment as above - rather than calling these based on how they came to be a files artifact (which is fairly irrelevant) seems like we'd instead want to name them based on what's inside of them